openshift-knative-operator/pkg/common/deployment.go

```diff
@@ -0,0 +1,73 @@
+package common
+
+import (
+	mf "github.com/manifestival/manifestival"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// InjectEnvironmentIntoDeployment injects the specified environment variables into the
+// specified deployment/container.
+// Note: We're not deleting empty environment variables and instead set them to empty
+// string. Three-way-merging of the deployment drops the update otherwise.
+func InjectEnvironmentIntoDeployment(deploymentName, containerName string, env map[string]string) mf.Transformer {
+	return transformDeployment(deploymentName, func(deploy *appsv1.Deployment) error {
+		containers := deploy.Spec.Template.Spec.Containers
```

Comment:
Are we planning to modify the deployment in other ways in the future? I am thinking about having this high-order func vs just collapsing this with the transformDeployment body anyway.

Reply:
I had the collapsed version before and this is imo nicer to read in that the annoying bits are out of the codepath I want to read.